packages/legend-query-builder/src/components/__test-utils__/QueryBuilderComponentTestUtils.tsx

```diff
@@ -49,6 +49,56 @@ import {
   TEST__LegendApplicationPluginManager,
   TEST__getGenericApplicationConfig,
 } from '../../stores/__test-utils__/QueryBuilderStateTestUtils.js';
+import { STYLE_PREFIX, STYLE_PREFIX__DARK } from '@finos/legend-art';
+import { expect } from '@jest/globals';
+
+export const selectFromCustomSelectorInput = (
+  selectorContainer: HTMLElement,
+  optionText: string,
+  lightMode?: boolean,
+): void => {
+  const selectorContainerClassName =
+    '.' + `${lightMode ? STYLE_PREFIX : STYLE_PREFIX__DARK}__value-container`;
+  const selectorInputValue =
+    '.' + `${lightMode ? STYLE_PREFIX : STYLE_PREFIX__DARK}__single-value`;
+
+  let foundOptionText = false;
+  const selector = selectorContainer.querySelector(
+    selectorContainerClassName,
+  ) as HTMLSelectElement;
+
+  fireEvent.keyDown(selector, { key: 'ArrowDown' });
+  fireEvent.keyDown(selector, { key: 'Enter' });
+  expect(selector.querySelector(selectorInputValue)).not.toBeNull();
+  const firstOptionValue = guaranteeNonNullable(
+    selector.querySelector(selectorInputValue),
+  ).textContent;
+  let currentOptionValue = firstOptionValue;
+  let numofPress = 1;
+
+  // Since the selector value can't be directly changed to the desired option value through fireEvent.change(),
+  // we need to repeatedly press the ArrowDown key to select it. However, each time we restart this process,
+  // it starts from the beginning of the dropdown list, creating a circular pattern.
+  // To prevent a loop, we must keep track of the number of times we press the key to select other options
+  // and ensure we don't return to the initial option.
+  while (
+    (!foundOptionText && currentOptionValue !== firstOptionValue) ||
+    numofPress === 1
+  ) {
+    for (let i = 0; i < numofPress + 1; i++) {
+      fireEvent.keyDown(selector, { key: 'ArrowDown' });
+    }
+    fireEvent.keyDown(selector, { key: 'Enter' });
+    expect(selector.querySelector(selectorInputValue)).not.toBeNull();
+    currentOptionValue = guaranteeNonNullable(
+      selector.querySelector(selectorInputValue),
+    ).textContent;
+    if (currentOptionValue === optionText) {
+      foundOptionText = true;
+    }
+    numofPress++;
+  }
```

Comment:
what happens if we don't find the option. what is the contract. we throw an error message saying it is not found ?